Added alternative unit minutes/km to speed in addition to km/h #130

abarber7 opened the pull request.

McPringle commented on src/test/java/app/gpx_animator/KmHrMinPerKmTest.java:
```diff
@@ -0,0 +1,58 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+public class KmHrMinPerKmTest {
+
+    private String calcKmPerHrToMinPerKm(final long speed) {
+        double minPerKm = speed;
+        double mins = (60 / minPerKm);
+        double secs = (mins - (int) mins) * 60;
+        double tenS = 10;
+        if (minPerKm != 0) {
+            if (secs == 0) {
+                return String.format("%d km/h | " + (int) mins + ":00 minutes/km", speed); //NON-NLS
+            } else if (secs >= tenS) {
+                return String.format("%d km/h | " + (int) mins + ":" + (int) secs + " minutes/km", speed); //NON-NLS
+            } else {
+                return String.format("%d km/h | " + (int) mins + ":0" + (int) secs + " minutes/km", speed); //NON-NLS
+            }
+        } else {
+            return String.format("%d km/h | 0:00 minutes/km", speed); //NON-NLS
+        }
+    }
```
This method looks identical to the same method in the ´Renderer´ class. In this case you test the method in the test class, not the method in the ´Renderer´ class.

McPringle commented:
Thank you very much for your PR, @abarber7. I would like to see an option in the GUI and on the command line to define the unit. There could be more than these two, like "mph" or "nm" (nautical miles). I'll merge your PR, do some refactorings and add the UI stuff.

abarber7 commented:
Hello,
I have Added "mph" and a "nm" (nautical miles) option.

Added alternative unit minutes/km to speed in addition to km/hr #130
Added alternative unit minutes/mile to speed in addition to mile/hr #130
Added alternative unit minutes/nm to speed in addition to nm/hr #130

McPringle commented:
Thank you very much for your work!

McPringle commented:
The ´Renderer´ class is very huge. So I decided to extract the speed calculation to a separate ´SpeedUtil´ class. With time, I hope that the ´Renderer´class will become clearer. Second, I created a ´SpeedUnit´ enum to collect all the speed units. That makes it easier to add a new unit (just one place to add it).

Again, thank you very much for your contribution. :-)
